test/trans/Tests/DropInferenceSpec.hs

```diff
@@ -0,0 +1,89 @@
+{-# LANGUAGE QuasiQuotes #-}
+-- | Sanity check test suite for equivalent and unequivalent programs.
+module Tests.DropInferenceSpec where
+import           Sslang.Test
+
+import qualified Front
+import           Front.Ast
+import qualified IR
+import           IR.DropInference               ( insertDropsProgram )
+import qualified IR.IR                         as I
+import           IR.IR
+import           IR.LambdaLift                  ( liftProgramLambdas )
+import           IR.Types.Poly                 as Poly
+parseLift :: String -> Pass (I.Program Poly.Type)
+parseLift s =
+  Front.run def s
+    >>= IR.lower def
+    >>= IR.ann2Class def
+    >>= IR.class2Poly def
+    >>= insertDropsProgram
+
+spec :: Spec
+spec = do
+  it "drop inference in named let binding" $ do
+    let undropped = parseLift [here|
+        top = 
+            let a: Int = 5
+                b: Int = 10
+            a + b
+        |]  
+    print $ runPass undropped
+    --undropped `shouldPassAs` dropped
```

Comment:
With #88 merged in, you should now be able to write:

```haskell
-- at top:
import IR.SubstMagic ( substMagic )
import Data.Data ( Proxy(..) )
-- later
let dropped = substMagic (Proxy :: I.Program Poly.Type) $ parseLift [here|
      top =
        let a: Int = 5
        let b: Int = 10
      let anon0 = a + b
      drop b
      drop a
      anon0
    |]
```

with luck, that will Just Work, but we can fix any bugs we find as we go.